rust/crd/src/affinity.rs

```diff
@@ -0,0 +1,177 @@
+
+pub fn get_affinity(cluster_name: &str, role: &AirflowRole) -> StackableAffinityFragment {
+    let affinity_between_cluster_pods = affinity_between_cluster_pods(APP_NAME, cluster_name, 20);
+    let affinity_between_role_pods =
+        affinity_between_role_pods(APP_NAME, cluster_name, &role.to_string(), 70);
+
+    StackableAffinityFragment {
+        pod_affinity: Some(PodAffinity {
+            preferred_during_scheduling_ignored_during_execution: Some(vec![
+                affinity_between_cluster_pods,
+            ]),
+            required_during_scheduling_ignored_during_execution: None,
+        }),
+        pod_anti_affinity: Some(PodAntiAffinity {
+            preferred_during_scheduling_ignored_during_execution: Some(vec![
+                affinity_between_role_pods,
+            ]),
+            required_during_scheduling_ignored_during_execution: None,
+        }),
+        node_affinity: None,
+        node_selector: None,
+    }
+}
+#[cfg(test)]
+mod tests {
+
+    use rstest::rstest;
+    use std::collections::BTreeMap;
+
+    use stackable_operator::{
+        commons::affinity::{StackableAffinity, StackableNodeSelector},
+        config::fragment::validate,
+        k8s_openapi::api::core::v1::{
+            NodeAffinity, NodeSelector, NodeSelectorRequirement, NodeSelectorTerm,
+        },
+        kube::runtime::reflector::ObjectRef,
+        role_utils::RoleGroupRef,
+    };
+
+    use crate::{AirflowCluster, AirflowRole};
+
+    use super::get_affinity;
+
+    #[rstest]
+    #[case(AirflowRole::Worker)]
+    #[case(AirflowRole::Scheduler)]
+    #[case(AirflowRole::Webserver)]
+    fn test_affinity_defaults(#[case] role: AirflowRole) {
+        let input = r#"
+        apiVersion: airflow.stackable.tech/v1alpha1
+        kind: AirflowCluster
+        metadata:
+          name: airflow
+        spec:
+          image:
+            productVersion: 2.4.1
+            stackableVersion: 23.4.0-rc2
+          executor: CeleryExecutor
+          loadExamples: true
+          exposeConfig: false
+          credentialsSecret: simple-airflow-credentials
+          webservers:
+            roleGroups:
+              default:
+                replicas: 1
+          workers:
+            roleGroups:
+              default:
+                replicas: 2
+          schedulers:
+            roleGroups:
+              default:
+                replicas: 1
+        "#;
+        let airflow: AirflowCluster = serde_yaml::from_str(input).expect("illegal test input");
+
+        let rolegroup_ref = RoleGroupRef {
+            cluster: ObjectRef::from_obj(&airflow),
+            role: role.to_string(),
+            role_group: "default".to_string(),
+        };
+
+        let expected: StackableAffinity = validate(get_affinity("airflow", &role)).unwrap();
+
+        let affinity = airflow
+            .merged_config(&role, &rolegroup_ref)
+            .unwrap()
+            .affinity;
+
+        assert_eq!(affinity, expected);
```

Comment:
Same as in https://github.com/stackabletech/zookeeper-operator/pull/649#discussion_r1115366099

Reply:
done.